Update ietf-json-patch (#489)
decentralized-identity/sidetree: docs/spec/markdown/patches.md

#### `ietf-json-patch`

> The `ietf-json-patch` _Patch Action_ describes an update to a DID Document, using [IETF JSON Patch](https://tools.ietf.org/html/rfc6902). To construct a `ietf-json-patch` _Patch Action_, compose an object as follows:

Reviewer: Suggested language to make this a bit more general for Methods that may choose to patch against in intermediate, abstract representation of DID state:

"The `ietf-json-patch` _Patch Action_ describes an update to a DID's state, using [IETF JSON Patch](https://tools.ietf.org/html/rfc6902)."

Author: @csuwildcat can you make this a suggestion that I can actually approve and merge?

> The `ietf-json-patch` _Patch Action_ describes an update to a DID Document, using [IETF JSON Patch](https://tools.ietf.org/html/rfc6902). To construct a `ietf-json-patch` _Patch Action_, compose an object as follows:

Reviewer: ```suggestion
The `ietf-json-patch` Patch Action describes a mechanism for modifying a DID's state using [IETF JSON Patch](https://tools.ietf.org/html/rfc6902). To construct a `ietf-json-patch` _Patch Action_, compose an object as follows:
```
